docs/AdvancedNAS.md

```diff
@@ -0,0 +1,111 @@
+
+#### Install NFS on server machine
+First, install NFS server:
+```bash
+sudo apt-get install nfs-kernel-server
+```
+Suppose `/tmp/nni/shared` is used as the physical storage, then run:
+```bash
+sudo mkdir -p /tmp/nni/shared
+sudo echo "/tmp/nni/shared *(rw,sync,no_subtree_check,no_root_squash)" >> /etc/exports
+sudo service nfs-kernel-server restart
+```
+You can check if the above directory is successfully exported by NFS using `sudo showmount -e localhost`
+
+#### Install NFS on client machine
+First, install NFS client:
+```bash
+sudo apt-get install nfs-common
+```
+Then create & mount the mounted directory of shared files:
+```bash
+sudo mkdir -p /mnt/nfs/nni/
+sudo mount -t nfs 10.10.10.10:/tmp/nni/shared /mnt/nfs/nni
+```
+where `10.10.10.10` should be replaced by the real IP of NFS server machine in practice.
+
+### Example code for trial
+In our example, we assign each layer a `hash_id` to identify whether a previously trained model weight is sharable,and construct the tensorflow graph using `hash_id` as the variable scope name:
+```python
+with tf.variable_scope(p_graph.layers[i].hash_id, reuse=tf.AUTO_REUSE):
+    # generate tensorflow operators for p_graph.layers[i]
+    ...
+```
+With hashes of all the sharable layer fed as `shared_id` hyper parameter, we can automatically initialize all the sharable layer from the previous trained model:
+```python
+tf.init_from_checkpoint(param['restore_path'], dict(zip(param['shared_id'], param['shared_id'])))
+```
+Where `param` is retrieved from customized tuner with `nni.get_next_parameter()`. An example configuration is shown as follows:
+```json
+{
+    "shared_id": [
+        "4a11b2ef9cb7211590dfe81039b27670",
+        "370af04de24985e5ea5b3d72b12644c9",
+        "11f646e9f650f5f3fedc12b6349ec60f",
+        "0604e5350b9c734dd2d770ee877cfb26",
+        "6dbeb8b022083396acb721267335f228",
+        "ba55380d6c84f5caeb87155d1c5fa654"
+    ],
+    "graph": {
+        "layers": [
+            ...
+            {
+                "hash_id": "ba55380d6c84f5caeb87155d1c5fa654",
+                "is_delete": false,
+                "size": "x",
+                "graph_type": 0,
+                "output": [
+                    6
+                ],
+                "output_size": 1,
+                "input": [
+                    7,
+                    1
+                ],
+                "input_size": 2
+            },
+            ...
+        ]
+    },
+    "restore_dir": "/mnt/nfs/nni/ga_squad/87",
+    "save_dir": "/mnt/nfs/nni/ga_squad/95"
+}
+```
+
+### Tuner customization for sharing policy
+We recommend implementing sharing policy for customized tuner through the calculation of `Layer.hash_id`. In our example, a layer is sharable iff. the configurations of the layer itself and all its previous layers are not changed. For details, see `Layer.update_hash` and `Graph.update_hash` function in [graph.py](../examples/tuners/weight_sharing/ga_customer_tuner/graph.py)
```

Comment:
what's the 'Layer.hash_id'? does all the customized tuner need them? what if I don't have concept of "layer" in my case?